tests/e2e/specs/miscellaneous/WorkspaceRunTimeout.spec.ts

```diff
@@ -0,0 +1,112 @@
+	const projectAndFileTests: ProjectAndFileTests = e2eContainer.get(CLASSES.ProjectAndFileTests);
+	const loginTests: LoginTests = e2eContainer.get(CLASSES.LoginTests);
+	const dashboard: Dashboard = e2eContainer.get(CLASSES.Dashboard);
+	const workspaces: Workspaces = e2eContainer.get(CLASSES.Workspaces);
+	const driverHelper: DriverHelper = e2eContainer.get(CLASSES.DriverHelper);
+	const cheCodeLocatorLoader: CheCodeLocatorLoader = e2eContainer.get(CLASSES.CheCodeLocatorLoader);
+	const webCheCodeLocators: Locators = cheCodeLocatorLoader.webCheCodeLocators;
+	const kubernetesCommandLineToolsExecutor: KubernetesCommandLineToolsExecutor = e2eContainer.get(
+		CLASSES.KubernetesCommandLineToolsExecutor
+	);
+	const shellExecutor: ShellExecutor = e2eContainer.get(CLASSES.ShellExecutor);
+	const testWorkspaceUtil: ITestWorkspaceUtil = e2eContainer.get(TYPES.WorkspaceUtil);
+
+	const stackName: string = 'Empty Workspace';
+	const cheClusterName: string = 'devspaces';
+	let runTimeoutValue: number = 0;
+
+	async function checkDialogButton(buttonName: string): Promise<void> {
+		await driverHelper.waitVisibility(By.xpath(`//div[@class='dialog-buttons']//a[text()='${buttonName}']`));
+	}
+
+	suiteSetup(function (): void {
+		kubernetesCommandLineToolsExecutor.loginToOcp();
+		shellExecutor.executeCommand('oc project openshift-devspaces');
+
+		// get current value of spec.devEnvironments.secondsOfRunBeforeIdling
+		runTimeoutValue = Number(
+			shellExecutor.executeCommand(
+				`oc get checluster/${cheClusterName} -o "jsonpath={.spec.devEnvironments.secondsOfRunBeforeIdling}"`
+			)
+		);
+
+		// set spec.devEnvironments.secondsOfRunBeforeIdling to 60
+		shellExecutor.executeCommand(
+			`oc patch checluster ${cheClusterName} --type=merge -p '{"spec":{"devEnvironments":{"secondsOfRunBeforeIdling": 60}}}'`
+		);
+	});
+
+	suiteTeardown(function (): void {
+		// restore spec.devEnvironments.secondsOfRunBeforeIdling to original value
+		shellExecutor.executeCommand(
+			`oc patch checluster ${cheClusterName} --type=merge -p '{"spec":{"devEnvironments":{"secondsOfRunBeforeIdling": ${runTimeoutValue}}}}'`
+		);
+	});
+
+	suiteSetup('Login', async function (): Promise<void> {
+		await loginTests.loginIntoChe();
+	});
+
+	test(`Create and open new workspace, stack:${stackName}`, async function (): Promise<void> {
+		await workspaceHandlingTests.createAndOpenWorkspace(stackName);
+		await workspaceHandlingTests.obtainWorkspaceNameFromStartingPage();
+	});
+
+	test('Wait workspace readiness', async function (): Promise<void> {
+		registerRunningWorkspace(WorkspaceHandlingTests.getWorkspaceName());
+		await projectAndFileTests.waitWorkspaceReadinessForCheCodeEditor();
+		await projectAndFileTests.performTrustAuthorDialog();
+	});
+
+	test('Wait run timeout dialog and check Dialog buttons', async function (): Promise<void> {
+		await driverHelper.waitVisibility(webCheCodeLocators.Dialog.details, TIMEOUT_CONSTANTS.TS_SELENIUM_START_WORKSPACE_TIMEOUT);
```

Comment:
The test is expected to check that timeout dialog to appear just after `secondsOfRunBeforeIdling` timeout succeeded, isn't it?

In that case `TIMEOUT_CONSTANTS.TS_SELENIUM_START_WORKSPACE_TIMEOUT` is not suitable in here, IMHO.

Reply:
thanks for the fix